---
title: "R Notebook"
output: html_notebook
---

```{python}
awk '{print $4}' /Users/henry/Documents/GitHub/Final/Gene_Counts/feec3523-951e-4541-8664-2a71adfdecb5.rna_seq.augmented_star_gene_counts.tsv > /Users/henry/Documents/GitHub/Final/Gene_Counts_Merged/feec.txt
```

```{bash}
paste gene_id.txt 00bb.txt 0c7c.txt 1fbb.txt 002d.txt 47b1.txt 5d3a.txt 7d5d.txt 7f20.txt 7fc4.txt 9ed8.txt 13db.txt 033f.txt 65ac.txt 67be.txt 78b8.txt 90fe.txt 92aa.txt 875e.txt 1536.txt 3330.txt 6633.txt 6813.txt 6855.txt 9077.txt a5f5.txt a66b.txt afa6.txt c9b3.txt d5d8.txt d92e.txt e1f5.txt ea66.txt f98d.txt f658.txt feec.txt > Merged_Gene_Counts.txt
```

```{r}
awk '{print $1 "," $7}' /Users/henry/Documents/GitHub/Final/pancrease_sample/exposure.tsv > /Users/henry/Documents/GitHub/Final/pancrease_sample/exposure.txt
```

```{r}
awk '{print $7}' /Users/henry/Documents/GitHub/Final/pancrease_sample/exposure.tsv > /Users/henry/Documents/GitHub/Final/pancrease_sample/sample2.tsv
```

```{r}
setwd('/Users/henry/Documents/GitHub/Final')

# Read in the matrix
count_matrix <- read.delim("/Users/henry/Documents/GitHub/Final/Gene_Counts_Merged/Merged_Gene_Counts.txt", header=T, sep="\t")

# Change column #1 into row name and delete the column #1 after that
row.names(count_matrix) <- count_matrix$gene_id
count_matrix <- count_matrix[-c(1)]

# Read in the sample sheet
sampletable <- read_tsv('/Users/henry/Documents/GitHub/Final/pancrease_sample/sample.tsv')

row.names(sampletable) <- sampletable$case_id

sampletable$alcohol_history <- as.factor(sampletable$alcohol_history)
```

```{r}
library(tibble)
library(tidyverse)
library(apeglm)
library(ggplot2)
library(vsn)
library(pheatmap)
library(ReportingTools)
library(DESeq2)
```

```{r}
DES_dataset <- DESeqDataSetFromMatrix(countData = count_matrix,
                                         colData = sampletable,
                                         design = ~ alcohol_history)
```

```{r}
nrow(DES_dataset)
```

```{r}
# Filtering to keep only rows that have at least 10 reads total
DES_dataset <- DES_dataset[rowSums(counts(DES_dataset)) > 10, ]

# Number of gene after filtering
nrow(DES_dataset)
```

```{r}
DES_dataset <- DESeq(DES_dataset)
```

```{r}
DES2Report <- HTMLReport(shortName = 'RNAseq_Analysis_with_DEseq2', title = 'Differential Expression Analysis in Pancrease', reportDirectory = "./reports")
publish(DES_dataset,DES2Report, pvalueCutoff=0.05, annotation.db="org.Mm.eg.db", factor = colData(DES_dataset)$alcohol_history, reportDir="./reports")
finish(DES2Report)
```

```{r}
result_table <- results(DES_dataset)
result_table
```

```{r}
resultLFC <- lfcShrink(DES_dataset, coef = "alcohol_history_Yes_vs_No", type = "apeglm")
```

```{r}
resultLFC
```

```{r}
plotMA(result_table, ylim=c(-2,2))
```

```{r}
plotMA(resultLFC, ylim=c(-2,2))
```

```{r}
result05 <- results(DES_dataset, alpha = 0.05)
summary(result05)
```

```{r}
sum(result05$padj < 0.05, na.rm = TRUE)
```

```{r}
plotCounts(DES_dataset, gene = which.min(result05$padj), intgroup = "alcohol_history")
```

```{r}
vsd <- vst(DES_dataset, blind = FALSE)
rld <- rlog(DES_dataset, blind = FALSE)
```

```{r}
head(assay(vsd), 3)
```

```{r}
ntd <- normTransform(DES_dataset)
meanSdPlot(assay(ntd))
```

```{r}
meanSdPlot(assay(vsd))
```

```{r}
meanSdPlot(assay(rld))
```

```{r}
select <- order(rowMeans(counts(DES_dataset,normalized=TRUE)),
                decreasing=TRUE)[1:20]
df <- as.data.frame(colData(DES_dataset)[,c("case_id", "alcohol_history")])
row.names(df) <- df$case_id
df <- df[-c(1)]
```

```{r}
pheatmap(assay(ntd)[select,], cluster_rows=FALSE, show_rownames=FALSE, cluster_cols=FALSE, annotation_col=df)
```

```{r}
pheatmap(assay(vsd)[select,], cluster_rows=FALSE, show_rownames=FALSE, cluster_cols=FALSE, annotation_col=df)
```

```{r}
pheatmap(assay(rld)[select,], cluster_rows=FALSE, show_rownames=FALSE, cluster_cols=FALSE,annotation_col=df)
```

