---
title: "R Notebook"
output: html_notebook
---

```{python}
awk '{print $2 , $7}' /Users/henry/Documents/GitHub/Final/pancrease_sample/exposure.tsv > /Users/henry/Documents/GitHub/Final/pancrease_sample/sample.txt
```

```{bash}
paste gene_id.txt TCGA-FB-AAQ0.txt TCGA-FB-AAPY.txt TCGA-3A-A9IU.txt TCGA-IB-7893.txt TCGA-HZ-7918.txt TCGA-FB-AAPQ.txt TCGA-IB-A6UF.txt TCGA-IB-7654.txt TCGA-F2-A8YN.txt TCGA-IB-7886.txt TCGA-IB-7646.txt TCGA-HZ-A49I.txt TCGA-3A-A9J0.txt TCGA-OE-A75W.txt TCGA-IB-A7LX.txt TCGA-XD-AAUL.txt TCGA-FB-AAPP.txt TCGA-FB-AAPZ.txt TCGA-IB-AAUP.txt TCGA-FB-AAQ1.txt TCGA-3E-AAAY.txt TCGA-IB-AAUR.txt TCGA-IB-A7M4.txt TCGA-S4-A8RM.txt TCGA-HZ-7925.txt TCGA-IB-A6UG.txt TCGA-IB-AAUM.txt TCGA-3A-A9I9.txt TCGA-IB-AAUU.txt TCGA-RB-AA9M.txt TCGA-YB-A89D.txt TCGA-IB-7647.txt TCGA-IB-AAUQ.txt TCGA-US-A77G.txt TCGA-HZ-A8P0.txt > Merged_Gene_Counts.txt
```

```{r}
awk '{print $2 "," $7}' /Users/henry/Documents/GitHub/Final/pancrease_sample/exposure.tsv > /Users/henry/Documents/GitHub/Final/pancrease_sample/sample.tsv
```

```{r}
awk '{print $2}' /Users/henry/Documents/GitHub/Final/pancrease_sample/exposure.tsv > /Users/henry/Documents/GitHub/Final/pancrease_sample/sample1.tsv
```

```{r}
setwd('/Users/henry/Documents/GitHub/Final')

# Read in the matrix
count_matrix <- read.delim("/Users/henry/Documents/GitHub/Final/Gene_Counts_Merged/Merged_Gene_Counts.txt", header=T, sep="\t")

# Change column #1 into row name and delete the column #1 after that
row.names(count_matrix) <- count_matrix$gene_id
count_matrix <- count_matrix[-c(1)]

# Read in the sample sheet
sampletable <- read_tsv('/Users/henry/Documents/GitHub/Final/pancrease_sample/sample.tsv')

row.names(sampletable) <- sampletable$case_submitter_id

sampletable$alcohol_history <- as.factor(sampletable$alcohol_history)
```

```{r}
library(tibble)
library(tidyverse)
library(apeglm)
library(ggplot2)
library(vsn)
library(pheatmap)
library(ReportingTools)
library(DESeq2)
```

```{r}
DES_dataset <- DESeqDataSetFromMatrix(countData = count_matrix,
                                         colData = sampletable,
                                         design = ~ alcohol_history)
```

```{r}
nrow(DES_dataset)
```

```{r}
# Filtering to keep only rows that have at least 10 reads total
DES_dataset <- DES_dataset[rowSums(counts(DES_dataset)) > 10, ]

# Number of gene after filtering
nrow(DES_dataset)
```

```{r}
DES_dataset <- DESeq(DES_dataset)
```

```{r}
DES2Report <- HTMLReport(shortName = 'RNAseq_Analysis_with_DEseq2', title = 'Differential Expression Analysis in Pancrease', reportDirectory = "./reports")
publish(DES_dataset,DES2Report, pvalueCutoff=0.05, annotation.db="org.Mm.eg.db", factor = colData(DES_dataset)$alcohol_history, reportDir="./reports")
finish(DES2Report)
```

```{r}
result_table <- results(DES_dataset)
result_table
```

```{r}
resultLFC <- lfcShrink(DES_dataset, coef = "alcohol_history_Yes_vs_No", type = "apeglm")
```

```{r}
resultLFC
```

```{r}
plotMA(result_table, ylim=c(-2,2))
```

```{r}
plotMA(resultLFC, ylim=c(-2,2))
```

```{r}
result05 <- results(DES_dataset, alpha = 0.05)
summary(result05)
```

```{r}
sum(result05$padj < 0.05, na.rm = TRUE)
```

```{r}
plotCounts(DES_dataset, gene = which.min(result05$padj), intgroup = "alcohol_history")
```

```{r}
vsd <- vst(DES_dataset, blind = FALSE)
rld <- rlog(DES_dataset, blind = FALSE)
```

```{r}
head(assay(vsd), 3)
```

```{r}
ntd <- normTransform(DES_dataset)
meanSdPlot(assay(ntd))
```

```{r}
meanSdPlot(assay(vsd))
```

```{r}
meanSdPlot(assay(rld))
```

```{r}
select <- order(rowMeans(counts(DES_dataset,normalized=TRUE)),
                decreasing=TRUE)[1:20]
df <- as.data.frame(colData(DES_dataset)[,c("case_submitter_id", "alcohol_history")])
df$case_submitter_id <- as.factor(df$case_submitter_id)
df <- df[-c(1)]
```

```{r}
pheatmap(assay(ntd)[select,], cluster_rows=FALSE, show_rownames=FALSE, cluster_cols=FALSE, annotation_col=df)
```

```{r}
pheatmap(assay(vsd)[select,], cluster_rows=FALSE, show_rownames=FALSE, cluster_cols=FALSE, annotation_col=df)
```

```{r}
pheatmap(assay(rld)[select,], cluster_rows=FALSE, show_rownames=FALSE, cluster_cols=FALSE,annotation_col=df)
```

```{r}
sampleDists <- dist(t(assay(vsd)))
DistMatrix <- as.matrix(sampleDists)
pheatmap(DistMatrix)
```

```{r}
plotPCA(vsd, intgroup="alcohol_history")
```

